python/cudf/cudf/tests/test_concat.py

```diff
@@ -1865,3 +1863,114 @@ def test_concat_mixed_list_types_error(s1, s2):
 
     with pytest.raises(NotImplementedError):
         cudf.concat([s1, s2], ignore_index=True)
+
+
+@pytest.mark.parametrize(
+    "axis",
+    [
+        pytest.param(
+            0,
+            marks=pytest.mark.xfail(
+                reason="concat dictionaries with axis=0 not implemented"
+            ),
+        ),
+        1,
+    ],
+)
+@pytest.mark.parametrize(
+    "d",
+    [
+        {"first": cudf.DataFrame({"A": [1, 2], "B": [3, 4]})},
```

Comment:
It's best if we can avoid creating instances of GPU objects (`cudf.DataFrame`) in the parametrize arguments. Those are executed at test _collection_ time rather than at test runtime, and make the test suite slow to launch due to a large number of small host-device copies. Let's defer construction using a pattern like this:

```python
@pytest.mark.parametrize(
    "d",
    [
        {"first": {"A": [1, 2], "B": [3, 4]}},
        # ...
    ],
)
def test_concat_dictionary(d, axis):
    # Convert dict-of-dicts to dict-of-DataFrames to avoid raw GPU objects in the parameters
    d = {k: cudf.DataFrame(v) for k, v in input.items()}

    result = cudf.concat(d, axis=axis)
    expected = cudf.from_pandas(
        pd.concat({k: df.to_pandas() for k, df in d.items()}, axis=axis)
    )
    assert_eq(expected, result)
```

Reply:
will do! let me know if you'd like me to clean up the other tests in this file in the same manner

Reply:
i passed in the reference to each class for the tests, let me know if this causes weirdness during test collection and i'll make a simple map.

Reply:
Yes, that’s totally fine. No need to refactor the rest of the file in this PR. A separate PR would be welcome if you’re interested.